File: 1567-maximum-number-of-vowels-in-a-substring-of-given-length/maximum-number-of-vowels-in-a-substring-of-given-length.cpp
class Solution {
public:
    int maxVowels(string s, int k) {
        int count=0;
        for(int i=0;i<k;i++){
            if(isVowel(s[i])){
                count++;
            }
        }
        int maxCount=count;
        for(int right=k;right<s.length();right++){
            if(isVowel(s[right-k])){
                count--;
            }
            if(isVowel(s[right])){
                count++;
            }
            maxCount=max(maxCount,count);
        }
        return maxCount;
    }
    bool isVowel(char ch){
        return string("AEIOUaeiou").find(ch)!=string::npos;
    }
};

File: 1567-maximum-number-of-vowels-in-a-substring-of-given-length/maximum-number-of-vowels-in-a-substring-of-given-length.java
class Solution {
    public int maxVowels(String s, int k) {
        int count=0;
        for(int i=0;i<k;i++){
            if(isVowel(s.charAt(i))){
                count++;
            }
        }
        int maxCount=count;
        for(int right=k;right<s.length();right++){
            if(isVowel(s.charAt(right-k))){
                count--;
            }
            if(isVowel(s.charAt(right))){
                count++;
            }
            maxCount=Math.max(maxCount,count);
        }
        return maxCount;
    }
    public boolean isVowel(char ch){
        return "AEIOUaeiou".indexOf(ch)!=-1;
    }
}